shared/connection.go

```diff
@@ -37,9 +51,15 @@ func MongoSession(uri string) *mgo.Session {
 	dialInfo.Direct = true // Force direct connection
 	dialInfo.Timeout = dialMongodbTimeout
 
+	err = opts.configureDialInfoIfRequired(dialInfo)
```

Comment:
I'm not a big fan of modifying values pass by reference. It's trickier to follow. Alternative:
```
err, dialServer := opts.dialServerFunc()
if err != nil {
	glog.Errorf("%s", err)
	return nil
}
dialInfo.DialServer = dialServer
```
It works so 👍